File: dw-58-59-code/Account.java

public class Account {

	private int balance;

	int getBalance() {
		return balance;
	}

	void setBalance(int balance) {
		this.balance = balance;
	}

	void deposit(int amt) {
		setBalance(getBalance() + amt);
	}

	void withdraw(int amt) {
		if (amt > getBalance())
			System.out.println("insufficeint amount");
		else
			setBalance(getBalance() - amt);
	}
}

File: php/phpproject/oop/Account.php
<?php
abstract class Account
{
    private $accountId;
    private $amt;
    private $name;
    private $location;

    public function __construct($accountId, $amt, $name, $location)
    {
        $this->amt = $amt;
        $this->accountId = $accountId;
        $this->name = $name;
        $this->location = $location;
    }

    public function getAccountDetails()
    {
        return "<p>account details <br> id " . $this->accountId . "<br>amount " . $this->amt . "<br>name" . $this->name . "<br>location" . $this->location;
    }

    public function getAmt(){
        return $this->amt ;
    }
    public function setAmt($amt){
     $this->amt=$amt ;
    }

    abstract public function deposit($currency);
    abstract public function withdraw($currency);
}
